src/plugins/core/header_visibility.ts

```diff
@@ -91,6 +92,16 @@ export class HeaderVisibilityPlugin extends CorePlugin {
     return this.hiddenHeaders[sheetId].COL[index];
   }
 
+  getAllVisibleHeaders(sheetId: UID, dimension: Dimension): HeaderIndex[] {
+    const visibleHeaders: HeaderIndex[] = [];
+    for (let i = 0; i < this.hiddenHeaders[sheetId][dimension].length; i++) {
```

Comment:
I think the allow dispatch should occur in this plugin rather than in core/sheet